lib/miq_automation_engine/engine/drb_remote_invoker.rb

```diff
@@ -63,10 +63,15 @@ def drb_cache_timeout
       1.hour
     end
 
+    def api_token
+      tm = TokenManager.new("api")
+      tm.gen_token("api", :userid => @workspace.ae_user.userid)
+    end
+
     # code building
 
-    def method_preamble(miq_uri, miq_id)
-      "MIQ_URI = '#{miq_uri}'\nMIQ_ID = #{miq_id}\n" << RUBY_METHOD_PREAMBLE
+    def method_preamble(miq_uri, miq_id, miq_api_token)
+      "MIQ_URI = '#{miq_uri}'\nMIQ_ID = #{miq_id}\nMIQ_API_TOKEN = '#{miq_api_token}'\n" << RUBY_METHOD_PREAMBLE
```

Comment:
@gmcculloug 
We can generate the token once in the beginning of the request.
We can't use a service method to create the token since in the future we would be running without DRb and we won't have access to the service object. By creating it once and sticking it the preamble it is available for the method to use and interact with the REST API on the backend